src/doc/ndarray_for_numpy_users/mod.rs

```diff
@@ -524,6 +524,18 @@
 //! </td></tr>
 //! </table>
 //!
+//! ## Type conversions
+//!
+//! `ndarray` does not currently have a `astype` function as `numpy`, but
+//! conversions between datatypes can be done with `mapv()`. The behavior of
+//! Rust's datatype conversion using the `as` keyword might be surprising for
+//! some `numpy` users, be sure to check out the type conversion example.
+//!
+//! NumPy | `ndarray` | Notes
+//! ------|-----------|------
+//! `a.astype(np.uint8)` | `unsafe{ a.mapv(|x| x.to_int_unchecked::<u8>()) }` | convert float array to new datatype without checking bounds, ignoring overflow
```

Comment:
I fear this might not be strongly worded enough, i.e. this does not just not check bounds but also imply UB if the values if out of bounds (or NaN). Personally, I would suggest strongly steering people away from these methods. Due to the above, I am also not sure whether the results will really be reliably compatible with NumPy due to unexpected compiler transformations?

Reply:
Thanks. Examples with `as` are ok. I agree with @adamreichold. Just skip mentioning `to_int_unchecked` unless you dedicate do it with care, I think I'd prefer not showing an example for it, just mention that it exists and link to its docs, if it's mentioned at all.

The perils of `to_int_unchecked` are **a lot** worse in Rust than anything you can do with `astype` in numpy.

Reply:
OK thank you for explaining and reinforcing this @adamreichold and @bluss  - so then I will remove the `f32::to_int_unchecked` and instead put the `a.mapv()` as rough equivalent of `a.astype()` with a comment mentioning the difference with saturating casts. With only one line, it might not make sense to have this as separate section. Would it fit into `## Array Manipulation` instead?

Reply:
I added a few more examples, not it might be fine as a section on its own.